onnxruntime/core/providers/nuphar/compiler/traverse_shape_infer.cc

```diff
@@ -82,7 +82,9 @@ Status ShapeInference(
 
   // perform shape inference using the topological order from ORT
   for (const NodeIndex& node_index : graph.GetNodesInTopologicalOrder()) {
-    const Node& node = *graph.GetNode(node_index);
+    const Node* p_node = graph.GetNode(node_index);
+    if(p_node == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "invalid node index");
+    const Node& node = *p_node;
```

Comment:
What model triggered this? This change is good, but the model probably needs update.

Reply:
Sorry, I can't remember. But I think it's always good to check if the pointer is NULL before dereferencing it (if you're not 100% sure).  A bad model shouldn't cause the program crash.